FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm

```diff
@@ -45,12 +47,22 @@
         /**
          Retain object and add it to returnValue.
          This operation can be unsafe since we are retaining object that could
-         be deallocated on other thread.
+         be deallocated on other thread. 
+         
+         When NSZombie enabled, we can find if object has been deallocated by checking its class name.
          */
-        returnValue.push_back(object);
+        if (fb_isNSZombieEnabled()) {
+          if (!fb_isZombieObject(object)) {
```

Comment:
What do you think of moving the `fb_isNSZombieEnabled()` call into `fb_isZombieObject()`? Since the former always has to be called before calling the latter, it makes it simpler for the caller here. Instead, this `if`-`if`-`else` logic becomes one `if`:

```objc
if (!fb_isZombieObject(object)) {
  returnValue.push_back(object);
}
```